domino.go

```diff
@@ -52,16 +52,34 @@ type ToValue interface {
 	ToDynamoDBValue() (bm interface{})
 }
 
-func serialize(item interface{}) (err error, av interface{}) {
+func serialize(item interface{}) (av map[string]*dynamodb.AttributeValue, err error) {
 	switch t := item.(type) {
 	case ToValue:
-		av = t.ToDynamoDBValue()
+		av, err = dynamodbattribute.MarshalMap(t.ToDynamoDBValue())
 	default:
 		av, err = dynamodbattribute.MarshalMap(item)
 	}
 	return
 }
 
+func marshal(m map[string]interface{}) (o map[string]*dynamodb.AttributeValue) {
+	if len(m) <= 0 {
+		return
+	}
+	o = map[string]*dynamodb.AttributeValue{}
+	for k, v := range m {
+		switch t := v.(type) {
+		case *dynamodb.AttributeValue:
+			o[k] = t
+		default:
+			a, _ := dynamodbattribute.Marshal(t)
```

Comment:
Unfortunately, this gets invoked during the construction of a query in the dsl. Propagating the error back would require complex delayed function logic that we have in other spots. For now I've chosen to ignore them, but probably better to panic. This should never throw however.